### PR TITLE
Sort by links the result from AllDefsPerPod

### PR DESCRIPTION
**Pull request by ewoutp**
This change makes sure that the maps returns by AllDefsPerPod are sorted in such a way that components linking to other components come after the components they link to.

### CODE REVIEW

**Comment by ewoutp** [comment]
@ZeissS RFR

**Comment by ZeissS** [comment]
I think this is something we can nicely write tests for.

**Comment by ewoutp** [comment]
> I think this is something we can nicely write tests for.

Yes we can!

Done

File: component_definitions.go [hunk 1]
```diff
@@ -336,6 +350,8 @@ func (nds *ComponentDefinitions) ComponentNames() ComponentNames {
 //   - group component definitions that share same pod
 //   - prevent duplicated lists, once a component definition is present in one
 //     list, it is not present in other lists.
+// The resulting maps are sorted such that components that link to other components are
+// found after the component that they link to.
 func (nds *ComponentDefinitions) AllDefsPerPod(names ComponentNames) ([]ComponentDefinitions, error) {
```
Comment (ZeissS) [inline component_definitions.go]: So this is like a groupByPod function? I know, replacing one name with another is not really helpful, but I am trying to get behind the goal here.

So, our ComponentDefinitions contains 3 components:
-  A (Pod "1")
-  B
-  C (Pod "1")

This function will return a slice with 2 items:
- A + C
- B

right?

Comment (ewoutp) [inline component_definitions.go]: Yep
It already did that, what I've added now is that the returned slices are order taking the links of components into account.

Comment (ZeissS) [inline component_definitions.go]: Sure, I just didn't know what the purpose was in the first place.

**Comment by ZeissS** [comment]
LGTM

**Pull request merged**